tst/EnergyPlus/unit/PlantCentralGSHP.unit.cc

```diff
@@ -200,3 +200,156 @@ TEST_F(EnergyPlusFixture, ChillerHeater_Autosize)
+         "    3.389817E-04,            !- Coefficient5 y**2",
+         "    -3.632396E-04,           !- Coefficient6 x*y",
+         "    4.44,                    !- Minimum Value of x",
+         "    12.78,                   !- Maximum Value of x",
+         "    12.78,                   !- Minimum Value of y",
+         "    29.44,                   !- Maximum Value of y",
+         "    ,                        !- Minimum Curve Output",
+         "    ,                        !- Maximum Curve Output",
+         "    Temperature,             !- Input Unit Type for X",
+         "    Temperature,             !- Input Unit Type for Y",
+         "    Dimensionless;           !- Output Unit Type",
+
+         "Curve:Biquadratic,",
+         "    ChillerHeaterHtgEIRFT,   !- Name",
+         "    0.2286246,               !- Coefficient1 Constant",
+         "    2.498714E-02,            !- Coefficient2 x",
+         "    -1.267106E-05,           !- Coefficient3 x**2",
+         "    9.327184E-03,            !- Coefficient4 y",
+         "    5.892037E-05,            !- Coefficient5 y**2",
+         "    -3.268512E-04,           !- Coefficient6 x*y",
+         "    4.44,                    !- Minimum Value of x",
+         "    15.56,                   !- Maximum Value of x",
+         "    35.0,                    !- Minimum Value of y",
+         "    57.22,                   !- Maximum Value of y",
+         "    ,                        !- Minimum Curve Output",
+         "    ,                        !- Maximum Curve Output",
+         "    Temperature,             !- Input Unit Type for X",
+         "    Temperature,             !- Input Unit Type for Y",
+         "    Dimensionless;           !- Output Unit Type",
+
+         " Curve:Cubic,",
+         "     ChillerHeaterClgEIRFPLR, !- Name",
+         "     0.0,                     !- Coefficient1 Constant",
+         "     1.22895,                 !- Coefficient2 x",
+         "     -0.751383,               !- Coefficient3 x**2",
+         "     0.517396,                !- Coefficient4 x**3",
+         "     0.2,                     !- Minimum Value of x",
+         "     1;                       !- Maximum Value of x",
+
+         "Curve:Cubic,",
+         "    ChillerHeaterHtgEIRFPLR, !- Name",
+         "    0.0,                     !- Coefficient1 Constant",
+         "    1.12853,                 !- Coefficient2 x",
+         "    -0.0264962,              !- Coefficient3 x**2",
+         "    -0.103811,               !- Coefficient4 x**3",
+         "    0.3,                     !- Minimum Value of x",
+         "    1;                       !- Maximum Value of x"
+
+        });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    // call the factory with a valid name to trigger reading inputs
+    state->dataPlantCentralGSHP->getWrapperInputFlag = true;
+
+    auto compPtr = PlantCentralGSHP::WrapperSpecs::factory(*state, "CHW_LOOP HEATPUMP1");
```

Comment:
This `compPtr` is unused and throwing a warning from the compiler.  I'm guessing you probably don't need to call the PlantComponent wrapper function anyway though, and instead just call the `GetWrapperInput` function instead (which the factory probably does anyway...)

Reply:
Defect file added.

Reply:
The change is also done now. It calling the "GetWrapperInput()' function directly, with some minor changes in the idf snippet, e.g. added a schedule.